test/library/postman-api-key.test.js

```diff
@@ -0,0 +1,196 @@
+const fs = require('fs'),
+    nock = require('nock'),
+    sinon = require('sinon'),
+    request = require('postman-request'),
+    COLLECTION = {
+        id: 'C1',
+        name: 'Collection',
+        item: [{
+            id: 'ID1',
+            name: 'R1',
+            request: 'https://postman-echo.com/get'
+        }]
+    },
+    VARIABLE = {
+        id: 'V1',
+        name: 'Variable',
+        values: [{
+            key: 'foo',
+            value: 'bar'
+        }]
+    };
+
+describe('newman.run postmanApiKey', function () {
+    before(function () {
+        nock('https://api.getpostman.com')
+            .persist()
+            .get(/^\/collections/)
+            .reply(200, COLLECTION);
+
+        nock('https://api.getpostman.com')
+            .persist()
+            .get(/^\/environments/)
+            .reply(200, VARIABLE);
+    });
+
+    after(function () {
+        nock.restore();
+    });
+
+    beforeEach(function () {
+        sinon.spy(request, 'get');
+    });
+
+    afterEach(function () {
+        request.get.restore();
+    });
+
+    it('should fetch collection via UID', function (done) {
+        newman.run({
+            collection: '1234-588025f9-2497-46f7-b849-47f58b865807',
+            postmanApiKey: '12345678'
+        }, function (err, summary) {
+            expect(err).to.be.null;
+            expect(request.get.calledOnce).to.be.true;
+
+            let requestArg = request.get.firstCall.args[0];
+
+            expect(requestArg).to.be.an('object').with.keys(['url', 'json', 'headers']);
+
+            expect(requestArg.url).to.be.a('string')
+                .that.is.equal('https://api.getpostman.com/collections/1234-588025f9-2497-46f7-b849-47f58b865807');
+
+            expect(requestArg.headers).to.be.an('object')
+                .that.has.property('X-Api-Key').to.equal('12345678');
+
+            expect(summary).to.be.an('object')
+                .that.has.property('collection').to.be.an('object')
+                .and.that.include({ id: 'C1', name: 'Collection' });
+
+            done();
+        });
+    });
+
+    it('should fetch environment via UID', function (done) {
+        newman.run({
+            collection: 'test/fixtures/run/single-get-request.json',
+            environment: '1234-931c1484-fd1e-4ceb-81d0-2aa102ca8b5f',
+            postmanApiKey: '12345678'
+        }, function (err, summary) {
+            expect(err).to.be.null;
+            expect(request.get.calledOnce).to.be.true;
+
+            let requestArg = request.get.firstCall.args[0];
+
+            expect(requestArg).to.be.an('object').with.keys(['url', 'json', 'headers']);
```

Comment:
Can be replaced with an includes assertion.

Reply:
`includes` doesn't work with objects.